### ISSUE TITLE
nonworking 'nls' method frmo base 'stats' package

### ISSUE THREAD

**Issue by donarus** [oracle/fastr#54]
The 'stats' package is one of the base packages preinstalled with R/FastR

there is a method NLS (non-linear least squares method)

In GNU R the following example work perfectly fine
```R
regressionData <- data.frame(
  x = c(1, 2, 3, 4),
  y = c(1.5, 3.3, 99, 16)
)
startingParams <- list(
  a = 1,
  q = 2
)
mNLS <- nls(formula = "y ~ a*x^q",
            data = regressionData,
            start = startingParams)
```

but in FastR it produces an error, although the input is valid.
```
Error in nlsModel(formula, mf, start, wts) :
  singular gradient matrix at initial parameter estimates
```

PS: If you want, I have multiple other examples of nonworking chunks from base packages, but I'm unsure where to track them

**Comment by steve-s** [comment]
Hi Tadeas, 

thank you for the report. It boils down to issue with `numericDeriv`:

```
env <- new.env()
env$a <- 1
env$q <- 2
env$x <- c(1,2,3,4)
env$y <- c(1.5, 3.3, 99.0, 16.0)
numericDeriv(quote(a * x^q), c("a", "q"), env)
```

gives a wrong result on FastR (gradient matrix is all zeroes). We are working on a fix.

> If you want, I have multiple other examples of nonworking chunks from base packages, but I'm unsure where to track them.

yes please, open GitHub issues or you can put them in one issue to make it simpler for you. 

It may be useful to distinguish issues that are blockers for your use-case and those that have a workaround. The most useful bug reports for us are those that point to an R builtin/primitive function or native function (called via `.Call`/`.Fortran` etc.) that gives different result on FastR, but any bug reports are welcome.

**Comment by steve-s** [comment]
Should be fixed in 86fafa2afe3ec8e1c401f901ae3dbfacc64bb191
